contracts/MemeFactory.sol

```diff
@@ -155,27 +160,32 @@ abstract contract MemeFactory {
 
     /// @dev Creates reference token + meme token LP and adds liquidity.
     /// @param memeToken Meme token address.
-    /// @param referenceTokenAmount Reference token amount.
+    /// @param nativeTokenAmount Native token amount.
     /// @param memeTokenAmount Meme token amount.
     /// @return pair reference token + meme token LP address.
     /// @return liquidity Obtained LP liquidity.
     function _createUniswapPair(
         address memeToken,
-        uint256 referenceTokenAmount,
+        uint256 nativeTokenAmount,
         uint256 memeTokenAmount
     ) internal returns (address pair, uint256 liquidity) {
-        // Create the LP
-        pair = IUniswap(factory).createPair(referenceToken, memeToken);
+        _wrap(nativeTokenAmount);
 
+        // TODO Check that this LP token doesn't exist
```

Comment:
It cannot exist. The token is only available in the factory before unleashing

Reply:
https://github.com/Uniswap/v2-core/blob/master/contracts/UniswapV2Factory.sol#L23
Any can call this and create a pair

Reply:
Basically when the token is created this pair creation can be front-run